### PR TITLE
test: Skip flaky K8sServices NodePort test

### PR DESCRIPTION
**Pull request by twpayne**
As [suggested by](https://github.com/cilium/cilium/issues/12690#issuecomment-1007256986) @pchaigno in #12690.

Signed-off-by: Tom Payne <tom@isovalent.com>

### CODE REVIEW

**Review by pchaigno — APPROVED** [review]

File: test/k8sT/Services.go [hunk 1]
```diff
@@ -1024,10 +1024,11 @@ Secondary Interface %s :: IPv4: (%s, %s), IPv6: (%s, %s)`, helpers.DualStackSupp
 					})
 
 					SkipItIf(func() bool {
-						// Quarantine when running with the third node as it's
-						// flaky. See #12511.
-						return helpers.GetCurrentIntegration() != "" ||
-							(helpers.SkipQuarantined() && helpers.ExistNodeWithoutCilium())
+						// Quarantine under all circumstances, as it's flaky,
+						// especially when running with th third node. See
```
Comment (christarazi) [inline test/k8sT/Services.go]: Nit: `th` :)

**Review by christarazi — APPROVED** [review]

**Review by kkourt — APPROVED** [review]

**Comment by pchaigno** [comment]
@twpayne You'll need to rebase to catch https://github.com/cilium/cilium/pull/18403, which fixes the Go test.

File: test/k8sT/Services.go (hunk 1 above)
Comment (twpayne) [reply test/k8sT/Services.go]: Done, thanks.

**Comment by twpayne** [comment]
> You'll need to rebase to catch #18403, which fixes the Go test.

Done, thanks.

File: test/k8sT/Services.go [hunk 2]
```diff
@@ -1059,10 +1059,11 @@ Secondary Interface %s :: IPv4: (%s, %s), IPv6: (%s, %s)`, helpers.DualStackSupp
...
 					SkipItIf(func() bool {
-						// Quarantine when running with the third node as it's
-						// flaky. See #12511.
-						return helpers.GetCurrentIntegration() != "" ||
-							(helpers.SkipQuarantined() && helpers.ExistNodeWithoutCilium())
+						// Quarantine under all circumstances, as it's flaky,
+						// especially when running with the third node. See
+						// https://github.com/cilium/cilium/issues/12511 and
+						// https://github.com/cilium/cilium/issues/12690.
+						return helpers.GetCurrentIntegration() != "" || helpers.SkipQuarantined()
```
Comment (brb) [inline test/k8sT/Services.go]: This test case when running without the third node (i.e., the node which doesn't run cilium-agent) is useless, as it is intended to test Cilium's N/S loadbalancing for the particular configuration. The latter can be tested only by the third node.

**Review by brb — CHANGES_REQUESTED** [review]

File: test/k8sT/Services.go (hunk 2 above)
Comment (pchaigno) [reply test/k8sT/Services.go]: Why is it run when there is no third node then? Was that a mistake?

Comment (brb) [reply test/k8sT/Services.go]: I think the previous assumption was that KPR=strict was only when there existed the third node which is no longer true.

Comment (pchaigno) [reply test/k8sT/Services.go]: Ok. So we should now just skip it if there is no node without Cilium, regardless of whether it's quarantined?
```suggestion
						return helpers.GetCurrentIntegration() != "" || helpers.DoesNotExistNodeWithoutCilium() || helpers.SkipQuarantined()
```

Comment (brb) [reply test/k8sT/Services.go]: :+1:

Comment (twpayne) [reply test/k8sT/Services.go]: Thanks, PR updated.

**Pull request merged**